dll/win32/setupapi/cfgmgr.c

```diff
@@ -7548,7 +7548,15 @@ CM_Request_Device_Eject_ExW(
         return CR_INVALID_FLAG;
 
     if (pszVetoName == NULL && ulNameLength != 0)
+    {
         return CR_INVALID_POINTER;
+    }
+    else if (pszVetoName != NULL && ulNameLength == 0)
+    {
+        /* Windows 2003 SP2 ignores pszVetoName when ulNameLength is zero
+         * and behaves like when pszVetoName is NULL */
+        pszVetoName = NULL;
+    }
```

Comment:
Could this whole code be simplified as:
```suggestion
    /* Windows 2003 SP2 ignores pszVetoName when ulNameLength
     * is zero and behaves like when pszVetoName is NULL */
    if (ulNameLength == 0)
        pszVetoName = NULL;

    if (!pszVetoName && ulNameLength != 0)
        return CR_INVALID_POINTER;
```

Reply:
Simplified.